evm-contracts/src/v0.7/dev/LinkTokenReceiver.sol

```diff
@@ -58,7 +59,7 @@ abstract contract LinkTokenReceiver {
       // solhint-disable-next-line avoid-low-level-calls
       funcSelector := mload(add(data, 32))
     }
-    require(funcSelector == ORACLE_REQUEST_SELECTOR, "Must use whitelisted functions");
+    require(funcSelector == ORACLE_REQUEST_SELECTOR || funcSelector == OPERATOR_REQUEST_SELECTOR, "Must use whitelisted functions");
```

Comment:
Also, I think this modifier and the constants being in `LinkTokenReceiver` doesn't really make sense. There are other functions that use `onTokenTransfer` but don't use `oracleRequest` or `requestOracleData`.

My first guess at how this would work would be that this is an abstract contract where `validateTokenTransferAction(bytes4)` is defined by the implementer. That is where this action would go.

I know it was here before the PR, but it just seems like it doesn't belong. Would be great to squeeze in here, but we could break it out into another story.

Reply:
Created a virtual function `validateTokenTransferAction(bytes4)` within `LinkTokenReceiver`, which is overridden by the `Operator`. This is where the function selector check occurs.